src/libAtomVM/globalcontext.h

```diff
@@ -121,6 +122,10 @@ struct GlobalContext
     SpinLock env_spinlock;
 #endif
 
+#if HAVE_OPEN && HAVE_CLOSE
+    ErlNifResourceType *posix_fd_resource_type;
+#endif
+
```

Comment:
I am not really much informed about the availability of POSIX APIs on platforms we currently target or want to target in the future, but what I was wondering here was whether it would make sense to instead of declaring this field on the global context conditionally, should just declare this as a static (dare I say, global) variable in your posix_nifs.c?  

I know I had issues you helped me with with competing declarations of this GlobalContext structure, which resulted in inconsistent definitions of the type in different compilation units, and this manifested itself in strange bus errors at runtime which were hard to diagnose.  That is my one issue with this strategy, but I don't know if the tradeoff of the strategy I am suggesting is worse.

Reply:
I would like to allow several GlobalContext coexist at the same time and therefore avoid any global variable.

The alternative would be to embed this into platform_data.

Reply:
Indeed, very good point.  I know there are cases where we are already using a global XQueue (though maybe that went away with the SMP branch), which makes having multiple VM instances in the same process space a challenge, but I get the point.

Reply:
+1 for moving into platform data, though again, I plead ignorance about the extent of support for POSIX APIs in most embedded platforms.  Not sure if we ever plan to support any of those 10 cent microcontrollers that are coming out of Shengzhen these days.